dist/index.js

```diff
@@ -7416,6 +7436,8 @@ function getSource(settings) {
                     core.startGroup('Fetching submodules');
                     yield git.submoduleSync(settings.nestedSubmodules);
                     yield git.submoduleUpdate(settings.fetchDepth, settings.nestedSubmodules);
+                    yield git.submoduleReset(settings.nestedSubmodules);
+                    yield git.submoduleClean(settings.nestedSubmodules);
```

Comment:
Shouldn't Reset and Clean be called prior to Update? I think the current code will fail if what was previously a folder belonging to the current repo is now the location of a submodule

Reply:
Also, it seems like this cleaning should only occur if clean=true

Reply:
Good point for the `clean=true`!
I am not sure about the importance of the order but you are probably right and I believe it would be more coherent with what is done for the main git repository. 

However, this project seems quite dead, and I will therefore not spend more time on this before getting the assurance that this could really be merged...

Reply:
It's sad if it's dead, `actions/checkout@v2` is used probably in 99.99% of the git actions out there...

Reply:
I do agree and I'm also using it a lot.
But no response on this PR, nor any commit on the project since 3 month. That's much for a project that is so necessary !

Reply:
Should be done in the commits proposed by @AceCoderLaura that are added to the PR.